stat.c

```diff
@@ -0,0 +1,147 @@
+/* Copyright (C) 2023, Oracle and/or its affiliates. All rights reserved
+ *
+ * This file is part of libresource.
+ *
+ * libresource is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * libresource is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with libresource. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _RESOURCE_H
+#include "resource.h"
+#endif
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include "resvm.h"
+#include "resource_impl.h"
+#include <errno.h>
+#include <libgen.h>
+#include "stat.h"
+
+static char buffer[9000];
+
+static int populate_statcpu(struct cpu_stat *all_cpu, int cpu_num)
+{
+	char buf[36];
+	const char *cstr;
+
+	if (!all_cpu) {
+		return -1;
+	}
+	for (int i = 0; i < cpu_num; i++) {
+		sprintf(buf, "cpu%d", i);
```

Comment:
I still prefer this be an snprintf.  This is outside the 3 lines of usual patch context of the buf[36] definition, so a future patch could cause trouble that we wouldn't catch.